Linting

brettz9 opened the pull request:
- Breaking change: Require Node 8
- Refactoring: Use ES6 classes
- Travis: Use Node 10, 12, 14
- Linting: Add `eslint-config-ash-nazg` and apply (and remove jshint) to JS, Markdown, HTML
- Maintenance: Add `.editorconfig`
- Testing: Add nyc for coverage
- Docs (README): Clean-up
- npm: Add separate `mocha` (no-coverage) test script and add coverage to `test`
- npm: Add `package-lock.json`
- npm: Add recommended fields (contributors, homepage)
- npm: Change deprecated `licenses` with `license`
- npm: Add `lint` and `lintfix` scripts
- npm: Update link from http to https
- npm: Simplify test (full relative binary path not needed in npm scripts)
- npm: Update devDeps.

Add changes similar to those proposed for passport (its depender)

brettz9 committed:
- Breaking change: Require Node 8
- Refactoring: Use ES6 classes
- Travis: Add Node 11 and 12
- Linting: Add eslint-config-ash-nazg (and remove jshint) to JS, Markdown, HTML
- Maintenance: Add `.editorconfig`
- npm: Add `package-lock.json`
- npm: Add recommended fields (contributors, homepage)
- npm: Change deprecated `licenses` with `license`
- npm: Add `lint` and `lintfix` scripts
- npm: Update link from http to https
- npm: Update chai, mocha

adamhathcock commented:
I need to get on the ball and add Typescript definitions now :)

brettz9 commented:
Cool :)  I am trying to finish up a Promise PR if that may affect your plans...

And if we're pulling off the Band-Aid with a new Promises API anyways, I would also like to propose a new API so that function signatures or return objects like `ret, user` could be done as `{ret, user}` instead to allow for passing arguments and destructuring without needing to care about argument order.

adamhathcock commented:
I’m happy to wait for Promises.  I think optional paeans should be grouped into an object too.

rwky commented:
I'll check this once https://github.com/passport-next/passport/pull/19 is merged.

brettz9 commented:
I've updated the devDeps and linted/tested against these updates... As the single dependency of passport, I'd think this would be good to update to a new major version first so the next passport update could depend on it.

brettz9 commented:
But note that even with this PR, passport-strategy still has make files (as does passport-local)...

brettz9 committed:
- Linting: As per latest ash-nazg
- npm: Simplify test (full relative binary path not needed in npm scripts)
- npm: Update devDeps.

brettz9 committed:
- Travis: Set Node versions to 10, 12, 14
- npm: Bump engines and browserslist to Node 10

brettz9 committed:
- Testing: Add nyc for coverage
- npm: Add separate `mocha` (no-coverage) test script and add coverage to `test`

brettz9 committed:
- npm: Update devDeps.

brettz9 committed:
- npm: Bump package-lock

brettz9 committed:
- Docs (README): Clean-up

brettz9 committed:
- Docs (README): Clean-up

brettz9 committed:
- npm: Update devDeps (including swapping eslint-config-ash-nazg peerDep. and drop need for typescript)

brettz9 committed:
- Linting: Use ESLint 7 new processor API as now targeted by `eslint-plugin-markdown` and `eslint-plugin-jsdoc`
- npm: Update devDeps.

brettz9 committed:
- Linting: As per latest ash-nazg
- npm: Update devDeps.
- Testing: Avoid need for own bootstrap file

brettz9 committed:
- Breaking change: Drop Node 10
- Testing Check Node 16
